azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -661,6 +739,62 @@ def remove_extension_slice(self, extension_name):
                 except Exception as exception:
                     _log_cgroup_warning("systemctl stop failed (remove slice): {0}", ustr(exception))
 
+        def set_extension_services_cpu_memory_quota(self, services_list):
+            """
+            Each extension service will have name, systemd path and it's quotas.
+            This method ensures that drop-in files are created under service.d folder if quotas given.
+            ex: /lib/systemd/system/extension.service.d/11-CPUAccounting.conf
+            TODO: set cpu and memory quotas
+            """
+            if self.enabled() and services_list is not None:
+                for service in services_list:
+                    service_name = service.get('name', None)
+                    unit_file_path = service.get('path', None)
+                    if service_name is not None and unit_file_path is not None:
+                        files_to_create = []
+                        drop_in_path = os.path.join(unit_file_path, "{0}.d".format(service_name))
+                        drop_in_file_cpu_accounting = os.path.join(drop_in_path,
+                                                                   _DROP_IN_FILE_CPU_ACCOUNTING)
+                        files_to_create.append((drop_in_file_cpu_accounting, _DROP_IN_FILE_CPU_ACCOUNTING_CONTENTS))
+
+                        self.__create_all_files(files_to_create)
+
+                # reload the systemd configuration; the new unit will be used once the service restarts
+                try:
+                    logger.info("Executing systemctl daemon-reload...")
+                    shellutil.run_command(["systemctl", "daemon-reload"])
+                except Exception as exception:
+                    _log_cgroup_warning("daemon-reload failed (create service unit files): {0}", ustr(exception))
+
```

Comment:
there is an asymmetry between start_tracking_extension_services_cgroups and stop_tracking_extension_services_cgroups. The former just adds the service to the tracked cgroups, the latter removes if from the tracked cgroups **and** also removes the dropin files . This may lead to problems as the code evolves. You should probably move the removal of the files to another method.

Reply:
I can't recall why I did that way. Sure, I can modularize that piece.